vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -72,6 +78,35 @@
     private Registration validationRegistration;
     private Registration dataProviderListenerRegistration;
 
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(ListDataProvider<T> dataProvider) {
+        this.setDataProvider((DataProvider<T, ?>) dataProvider);
+        return getListDataView();
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(Collection<T> items) {
+        return setDataProvider(DataProvider.ofCollection(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(Stream<T> items) {
+        return setDataProvider(DataProvider.fromStream(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(T... items) {
+        return setDataProvider(DataProvider.ofItems(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> getListDataView() {
+        if (dataProvider instanceof ListDataProvider) {
```

Comment:
Maybe check for `null` and throw NPE with good message

Reply:
Done.